pymongo_inmemory/mongod.py

```diff
@@ -34,7 +35,8 @@ def clean_before_kill(signum, stack):
     exit()
 
 
-signal.signal(signal.SIGTERM, clean_before_kill)
+if threading.current_thread() is threading.main_thread():
```

Comment:
Could you add a comment here summarising why this is added? So that we don't remove it in the future.

Reply:
I've added a comment. I'll look into adding a test. 

I am unsure however of the side effects it might have as it might not be able to perform the clean when killed. For my usage personally it does not make a difference as it's for dev purposes and it will be a real MongoDB when in production. My problem in particular is that i'm starting the client in code that is run by an asyncio loop that is not running on the main thread.

Reply:
The cleanup part was a bit overkill, I agree, but the aim is to leave the system as close to pre-test as possible.

Thank you for the contribution! I'll wait until weekend if you want to add some tests, otherwise I'll merge it.